src/modules/action-button/action-button.component.spec.ts

```diff
@@ -0,0 +1,61 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { DebugElement } from '@angular/core';
+import { Router } from '@angular/router';
+import { expect } from '@blackbaud/skyux-builder/runtime/testing/browser';
+import { SkyModule } from '@blackbaud/skyux/dist/core';
+import { StacheGridModule } from '../grid/grid.module';
+
+import { StacheActionButtonComponent } from './action-button.component';
+
+describe('Action Button Component', () => {
+  let component: StacheActionButtonComponent;
+  let fixture: ComponentFixture<StacheActionButtonComponent>;
+  let debugElement: DebugElement;
+  let mockRouter: any;
+      class MockRouter {
+          public navigate = jasmine.createSpy('navigate');
+      }
+
+  beforeEach(() => {
+    mockRouter = new MockRouter();
+
+    TestBed.configureTestingModule({
+      declarations: [
+        StacheActionButtonComponent
+      ],
+      imports: [
+        SkyModule,
+        StacheGridModule
+      ],
+      providers: [{ provide: Router, useValue: mockRouter }]
+    })
+    .compileComponents();
+    fixture = TestBed.createComponent(StacheActionButtonComponent);
+    component = fixture.componentInstance;
+    debugElement = fixture.debugElement;
+  });
+
+  it('Should create an action button component.', () => {
+    expect(component).toExist();
+  });
+
+  describe('navigate', () => {
```

Comment:
<img width="442" alt="screen shot 2017-04-10 at 5 05 08 pm" src="https://cloud.githubusercontent.com/assets/20796436/24882689/f8606576-1e0f-11e7-97f0-3afdb97159a9.png">

Is nesting describe blocks inside the component to break down functionality not appropriate?